pkg/antctl/raw/upgrade/apistorage/command.go

```diff
@@ -0,0 +1,261 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"antrea.io/antrea/pkg/antctl/raw"
+)
+
+var example = strings.Trim(`
+  Upgrade all existing persisted resources to the latest API version
+  $ antctl upgrade api-storage
+
+  Upgrade persisted resources for a specified Antrea CRD to the latest API version
+  $ antctl upgrade api-storage --crd-list=antreaagentinfos.crd.antrea.io
+
+  Upgrade persisted resources for some specified Antrea CRDs to the latest API version
+  $ antctl upgrade api-storage --crd-list=egresses.crd.antrea.io,groups.crd.antrea.io
+`, "\n")
+
+type options struct {
+	k8sClient   client.Client
+	crdNameList []string
+}
+
+var opts *options
+
+func NewCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "api-storage",
+		Short:   "Upgrade existing persisted resources to the latest API version",
+		Example: example,
+		RunE:    runE,
+	}
+	o := &options{}
+	command.Flags().StringSliceVar(&o.crdNameList, "crd-list", nil, "")
+	opts = o
+
+	return command
+}
+
+func (o *options) complete(cmd *cobra.Command) error {
+	if o.k8sClient != nil {
+		return nil
+	}
+
+	scheme := runtime.NewScheme()
+	apiextinstall.Install(scheme)
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	o.k8sClient, err = client.New(kubeconfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runE(cmd *cobra.Command, _ []string) error {
+	if err := opts.complete(cmd); err != nil {
+		return err
+	}
+
+	crdNameListToUpgrade, err := getCRDNameListToUpgrade(cmd, opts.k8sClient, sets.New[string](opts.crdNameList...))
+	if err != nil {
+		return err
+	}
+
+	crdObjectsToUpgrade := getCRDListToUpgrade(cmd, opts.k8sClient, crdNameListToUpgrade)
+	for _, crd := range crdObjectsToUpgrade {
+		if err = upgradeCRDObjects(cmd, opts.k8sClient, crd); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to upgrade CRD %q: %v.\n", crd.Name, err)
+		}
+		if err = patchCRDStoredVersions(opts.k8sClient, crd); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to patch CRD %q status.storedVersion: %v.\n", crd.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// getAllAntreaCRDNames is used to get all CRD names whose suffix is "crd.antrea.io".
+func getAllAntreaCRDNames(k8sClient client.Client) (sets.Set[string], error) {
+	crdNameList := sets.New[string]()
+	crdList := &apiextv1.CustomResourceDefinitionList{}
+	if err := k8sClient.List(context.TODO(), crdList); err != nil {
+		return nil, fmt.Errorf("failed to get CRD list: %v", err)
+	}
+	for _, crd := range crdList.Items {
+		if strings.HasSuffix(crd.Name, "crd.antrea.io") {
+			crdNameList.Insert(crd.Name)
+		}
+	}
+
+	return crdNameList, nil
+}
+
+// getCRDNameListToUpgrade is used to get the name list of CRDs to upgrade.
```

Comment:
ditto, you may remove `is used to` for all methods' comments, this is unnecessary.

Reply:
Updated all related comments.